pkg/pillar/dpcreconciler/genericitems/dhcpcd.go

```diff
@@ -53,8 +54,26 @@ func (c Dhcpcd) Type() string {
 
 // Equal is a comparison method for two equally-named Dhcpcd instances.
 func (c Dhcpcd) Equal(other depgraph.Item) bool {
```

Comment:
Why create `Equal(other)` for `Dhcpd` struct, not for `Dhcpcd.DhcpConfig` ?

Reply:
Having the object and the argument be of the same type makes sense for an Equal function.
Currently the type of the argument is different. 
Don't know whether the type of both should be make Dhcpcd, or both changed to DhcpConfig. Either one would work for me.

Reply:
@eriknordmark 
The content of the Equal method was indeed a bit confusing. Type and the argument are actually the same: `Dhcpcd`
Changed code to make it clear.
```
func (c Dhcpcd) Equal(other depgraph.Item) bool {
	c2 := other.(Dhcpcd)
        ...
```
@uncleDecart 
Note that `Dhcpd` encapsulates `types.DhcpConfig` and adds two attributes for config item identification (adapter name+label in this case) - this is for [Reconciler](https://github.com/lf-edge/eve/tree/master/libs/reconciler)'s sake (need for the [Name method](https://github.com/lf-edge/eve/blob/master/libs/depgraph/depgraph_api.go#L116-L120)). Reconciler runs Equal for two instances of the same item, i.e. with _equal identification_, to determine if config changed and Modify needs to be called to apply changed config. So attributes used for item identification don't need to be compared in Equal, they are guaranteed to be the same. So while we need to implement Equal for `Dhcpd` (so that it implements [Item](https://github.com/lf-edge/eve/blob/master/libs/depgraph/depgraph_api.go#L109-L150) interface), it is only content of the embedded `DhcpConfig` that actually needs to be compared.